jetty-core/jetty-websocket/jetty-websocket-jetty-api/src/main/java/org/eclipse/jetty/websocket/api/annotations/WebSocket.java

```diff
@@ -15,50 +15,31 @@
 
 import java.lang.annotation.Documented;
 import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
-import org.eclipse.jetty.websocket.api.BatchMode;
-import org.eclipse.jetty.websocket.api.StatusCode;
+import org.eclipse.jetty.websocket.api.Session;
 
 /**
- * Tags a POJO as being a WebSocket class.
+ * <p>Annotation for classes to be WebSocket endpoints.</p>
  */
 @Documented
 @Retention(RetentionPolicy.RUNTIME)
-@Target(value =
-    {ElementType.TYPE})
+@Inherited
+@Target(value = ElementType.TYPE)
 public @interface WebSocket
 {
     /**
-     * The size of the buffer (in bytes) used to read from the network layer.
+     * <p>Returns whether demand for WebSocket frames is automatically performed
+     * upon successful return from methods annotated with {@link OnWebSocketConnect},
+     * {@link OnWebSocketFrame} and {@link OnWebSocketMessage}.</p>
+     * <p>If the demand is not automatic, then {@link Session#demand()} must be
+     * explicitly invoked to receive more WebSocket frames (both control and
+     * data frames, including CLOSE frames).</p>
+     *
+     * @return whether demand for WebSocket frames is automatic
      */
-    int inputBufferSize() default -1;
-
-    /**
-     * The maximum size of a binary message (in bytes) during parsing/generating.
-     * <p>
-     * Binary messages over this maximum will result in a close code 1009 {@link StatusCode#MESSAGE_TOO_LARGE}
-     */
-    int maxBinaryMessageSize() default -1;
-
-    /**
-     * The time in ms (milliseconds) that a websocket may be idle before closing.
-     */
-    int idleTimeout() default -1;
-
-    /**
-     * The maximum size of a text message during parsing/generating.
-     * <p>
-     * Text messages over this maximum will result in a close code 1009 {@link StatusCode#MESSAGE_TOO_LARGE}
-     */
-    int maxTextMessageSize() default -1;
```

Comment:
This ability to configure in the annotation is quite popular.
Why remove it?

Reply:
Because the annotation was reporting only 4 configuration parameters out of the 8 that can actually be configured.

Furthermore, I see annotation attributes more as something configured once at the class level (like the annotation is specifying, similar to implementing an interface) -- for example you cannot change decoders on-the-fly, or auto-demand, etc., while the setters can be invoked at any time on the session, so it makes less sense to specify them at the class level.